### DISCUSSION TITLE
Change outlayer from [n,h,w,c] to [n,c,h,w]

### DISCUSSION THREAD

**Discussion by fehlneran** [onnx/onnx#5445] [Q&A]
I wonder how I can change the output format of an already existing Onnx models, i.e. from [n,h,w,c] to [n,c,h,w].
Gladly also with tools of the Onnx Community?

**Comment by cbourjau** [comment]
Would the [`Transpose`](https://github.com/onnx/onnx/blob/main/docs/Operators.md#transpose) operator cover your use case? Here is an example using [Spox](https://github.com/Quantco/spox):

```python
ifrom onnx import ModelProto
import numpy as np
import spox.opset.ai.onnx.v18 as op
from spox import build, inline, argument, Tensor

def some_existing_model() -> ModelProto:
    a = argument(Tensor(np.float32, ("N", "H", "W", "C")))
    b = op.identity(a)
    return build(inputs={"a": a}, outputs={"b": b})


# Wrap existing model and fix order of output axes
a = argument(Tensor(np.float32, ("N", "H", "W", "C")))
(nhwc,) = inline(some_existing_model())(a).values()
nchw = op.transpose(nhwc, perm=(0, 3, 1, 2))

model_with_reordered_axes: ModelProto = build(inputs={"a": a}, outputs={"nchw": nchw})
```

The individual functions are all documented in Spox's docs but feel free to ask if you have questions.
